zcash_client_sqlite/src/wallet/sapling.rs

```diff
@@ -159,8 +168,10 @@ pub(crate) fn get_spendable_sapling_notes(
             -- select all the unscanned ranges involving the shard containing this note
             WHERE sapling_received_notes.commitment_tree_position >= unscanned.start_position
             AND sapling_received_notes.commitment_tree_position < unscanned.end_position_exclusive
-            -- exclude unscanned ranges above the anchor height which don't affect spendability
+            -- exclude unscanned ranges that start above the anchor height (they don't affect spendability)
             AND unscanned.block_range_start <= :anchor_height
+            -- exclude unscanned ranges that end below or on the wallet birthday
+            AND unscanned.block_range_end > :wallet_birthday
```

Comment:
For the same reason as before:
```suggestion
            AND (unscanned.block_range_end - 1) > :wallet_birthday
```